sqlmesh/core/macros.py

```diff
@@ -303,11 +304,10 @@ def columns_to_types(self, model_name: TableName | exp.Column) -> t.Dict[str, ex
     def get_snapshot(self, model_name: TableName | exp.Column) -> t.Optional[Snapshot]:
         """Returns the snapshot that corresponds to the given model name."""
         if isinstance(model_name, str):
-            key = model_name
-        else:
-            key = ".".join(part.name for part in model_name.parts)
+            model_name = exp.to_table(model_name, dialect=self.dialect)
 
-        return self._snapshots.get(key)
+        normalize_model_name = normalize_identifiers(model_name, dialect=self.dialect)
```

Comment:
shouldn't this be normalize_model_name?

Reply:
ah thanks I keep forgetting about these utilities. Done.

Reply:
There's actually a small issue with `normalize_model_name` in case we get a column actually, because tables are built out of columns using `exp.table_(*reversed(table.parts[:-1]))`, but in the case of a macro the `Column` is actually just a `Table` represented differently and we don't want to skip `this` (i.e. the last element in `parts`). Perhaps we should convert the `Column` into a `Table` first..

Reply:
Took a stab at addressing it like this in https://github.com/TobikoData/sqlmesh/pull/1761/commits/4c1086889257ad93a0a60d85c8268db7eea0bdf3

Reply:
then normalize_model_name must be bugged?

Reply:
It depends on the interface it's supposed to implement. If it wants to extract out & normalize the `catalog.db.table` part given a column like `catalog.db.table.col`, then it's not bugged. If it was meant to interpret a `Column` as a table, then yeah it's bugged. I bet it's the former though because the latter is unintuitive and not documented.

Reply:
Ok let me check

Reply:
It doesn't seem that way, some tests are breaking if I change the logic in `normalize_model_name`:

```
    def test_get_measure_and_dim_tables():
>       assert _get_measure_and_dim_tables(d.parse_one("SUM(a.x)"), "") == ("a", ())
E       AssertionError: assert ('a.x', ()) == ('a', ())
E         At index 0 diff: 'a.x' != 'a'
E         Use -v to get more diff
```

The code diff I get this with is

```diff
-        table = exp.table_(*reversed(table.parts[:-1]))  # type: ignore
-    else:
-        table = exp.to_table(table, dialect=dialect)
+        table = exp.table_(table.this, db=table.args.get("table"), catalog=table.args.get("db"))

-    return exp.table_name(normalize_identifiers(table, dialect=dialect))
+    table = exp.to_table(table, dialect=dialect)
+    return exp.table_name(
+        normalize_identifiers(exp.to_table(table, dialect=dialect), dialect=dialect)
+    )